lib/improver/tests/wind_direction/test_WindDirection.py

```diff
@@ -159,41 +237,121 @@ def test_converts_array(self):
+        self.assertIsInstance(result, Cube)
+        self.assertIsInstance(result.data, np.ndarray)
+        self.assertArrayAlmostEqual(result.data, self.expected_wind_mean)
+
+
 class Test_find_r_values(IrisTest):
     """Test the find_r_values function."""
 
+    def setUp(self):
+        """Initialise plugin and supply data for tests"""
+        self.plugin = WindDirection()
+
     def test_converts_single(self):
         """Tests that r-value correctly extracted from complex value."""
+        self.plugin.wdir_slice_mean = make_wdir_cube_222()[0][0][0]
         expected_out = 2.0
-        complex_in = 1.4142135624+1.4142135624j  # Complex for angle=45 and r=2
-        result = WindDirection().find_r_values(complex_in)
-        self.assertAlmostEqual(result, expected_out)
+        # Set-up complex values for angle=45 and r=2
+        self.plugin.wdir_mean_complex = 1.4142135624+1.4142135624j
+        self.plugin.find_r_values()
+        self.assertAlmostEqual(self.plugin.r_vals_slice.data, expected_out)
 
     def test_converts_array(self):
         """Test that code can find r-values from array of complex numbers."""
+        longitude = DimCoord(np.linspace(-180, 180, 36),
+                             standard_name='longitude', units='degrees')
+
+        cube = Cube(COMPLEX_ANGLES, standard_name="wind_from_direction",
+                    dim_coords_and_dims=[(longitude, 0)],
+                    units="degree")
+        self.plugin.wdir_slice_mean = cube
+        self.plugin.wdir_mean_complex = COMPLEX_ANGLES
         expected_out = np.ones(COMPLEX_ANGLES.shape, dtype=np.float32)
-        result = WindDirection().find_r_values(COMPLEX_ANGLES)
+        self.plugin.find_r_values()
+        result = self.plugin.r_vals_slice.data
         self.assertIsInstance(result, np.ndarray)
         self.assertArrayAlmostEqual(result, expected_out)
 
 
 class Test_calc_confidence_measure(IrisTest):
     """Test the calc_avg_dist_mean function returns confidence values."""
 
+    def setUp(self):
+        """Initialise plugin and supply data for tests"""
+        self.plugin = WindDirection()
+        self.plugin.wdir_complex = WIND_DIR_COMPLEX
+        self.plugin.realization_axis = 0
+        self.plugin.r_vals_slice = make_wdir_cube_222()[0]
+        self.plugin.r_vals_slice.data = WIND_DIR_R_VALS
+        self.plugin.wdir_slice_mean = make_wdir_cube_222()[0]
+
     def test_returns_confidence(self):
         """First element has two angles directly opposite (90 & 270 degs).
         Therefore the calculated mean angle of 180 degs is basically
         meaningless. This code calculates a confidence measure based on how
         far the individual ensemble realizationss are away from
         the mean point."""
 
-        expected_out = np.array([[0.0, 0.95638061],
-                                 [0.91284426, 0.91284426]])
+        expected_out = np.array([[0.0, 0.956422],
+                                 [0.913176, 0.913176]])
```

Comment:
Why have these values changed?

Reply:
Sorry, I should have commented on this before.
I believe this is because the original test used complex numbers where the radius was not 1 so my correction to the plugin to rescale the complex mean to have a radius of 1 causes slightly different behaviour.

Reply:
But the code has (and still has) radius=1 hardcoded as the default value. So there shouldn't be a difference between the old and new code in this regard?
This looks like a rounding issue. Often if you're defining new arrays they go to float64 by default. In my code I tried to keep everything float32 to speed up calculations. Perhaps one of the new arrays you defined was float64 and all the maths was subesquently upgraded to float 64?